From a8cb48a6d6d851ef0e0f45224f5d94cffa0504b1 Mon Sep 17 00:00:00 2001
From: "Wu, Hao" <hao.wu@intel.com>
Date: Fri, 27 Mar 2015 11:32:48 +0800
Subject: [PATCH] dwc3: disable watchdog workaround if ULPI PHY is not used

Watchdog is added for problem with TI ULPI PHY, but other type
USB PHY doesn't need such workaround. This watchdog workaround
has side effects that cause failures in USB compliance test (e.g
USBCV and PET tests for OTG compliance test). So just disable the
watchdog workaround if ULPI PHY is not used. (e.g for Cherrytrail,
ULPI PHY is not used).

Change-Id: I95b64876cd226f833e40a105d6fd2b813b6a3ffd
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-15159
Signed-off-by: Wu, Hao <hao.wu@intel.com>
---
 drivers/usb/dwc3/gadget.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index c43f157..7ee85a8 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -2338,7 +2338,8 @@ static void dwc3_gadget_reset_interrupt(struct dwc3 *dwc)
 	 * In case TUSB121x phy is not used, this watchdog will create no harm
 	 * since we're not expecting it to be ever triggered.
 	 */
-	dwc3_gadget_kick_dog(dwc);
+	if (dwc->ulpi_phy)
+		dwc3_gadget_kick_dog(dwc);
 }
 
 static void dwc3_update_ram_clk_sel(struct dwc3 *dwc, u32 speed)
-- 
1.9.1

